manager/dispatcher/dispatcher.go

```diff
@@ -216,6 +217,9 @@ func (d *Dispatcher) Run(ctx context.Context) error {
 
 	defer cancel()
 	d.ctx, d.cancel = context.WithCancel(ctx)
+	ctx = d.ctx
```

Comment:
From https://tip.golang.org/pkg/context/, `Do not store Contexts inside a struct type`. This usage seems weird to me.